import React, { Component } from 'react'
import PipePic from '../../../media_files/flowchart.png';

export class Descriptions extends Component {

    render() {

        return (
            <div className="box">
                <img style={{width:'100%'}} src={PipePic}/>
                <div className={'paragraph_desc'}>
                    <h1 className={'title_description'} id="Fech pipeline">Fech pipeline </h1>
                    <p>
                        Fech pipeline pulls the Cellranger data of experiments according to list of IDs from lustre to irods. Subsequently a multiple metadata metrics are generated by default which include estimated cell counts, sequencing performance and other. This metadata can be retrieved directly from our scRNA website per run.
                        As part of the sequencing data is processed with <a href={'https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger'}>Cell Ranger</a>. This also produces an interactive htmls integrated here in the website.</p>
                </div>
                <div className={'paragraph_desc'}>
                    <h1 className={'title_description'} id="Fech">Cellbender</h1>
                    <p>
                        <a href={'https://github.com/broadinstitute/CellBender'}>Cellbender</a> is used to eliminate technical artefacts from the single cell RNA seq datasets. As part of the pipeline we have addapted the pipeline and use it to remove abbient RNA and random barcode swapping.
                    </p>
                </div>
                <div className={'paragraph_desc'}>
                    <h1 className={'title_description'} id="Deconvolution">Deconvolution</h1>
                    <p>
                        For deconvolution <a href={'https://github.com/single-cell-genetics/cellSNP'}>CellSnp</a> and <a href={'https://github.com/single-cell-genetics/vireo'}>Vireo</a> are used to split the multiplexed donors. The output generates a plots per pool representing donors and shows the number of cells per donor after deconvolution.
                </p>
                </div>
                <div className={'paragraph_desc'}>
                    <h1 className={'title_description'} id="QC metrics">QC Metrics</h1>
                    <p>

                        This QC pipeline, developed almost exclusively by Leland Taylor and Monika Krzak, is the third piece in a trio of pipelines designed to work sequentially in order to ingest (continuous) batches of Cellranger data generated at Sanger (cellranger runs are placed by Sequencing into Irods) for the Cardinal project.</p>
                    <p>
                        <p>
                            This QC pipeline runs on the Sanger LSF farm, on a “batch” of (deconvoluted) single Cell data, which it merges, normalizes, QCs, annotates and clusters in a series of Nextflow tasks, producing numerous plots and output files along the way.
                            The input “batch” must be a set of scRNA data for multiple samples/donors, i.e. a list of individual deconvoluted donors scRNA data (deconvoluted from cellranger experiments(s) by the HGI Vireo deconvolution pipeline).</p>
                    </p>
                </div>
                <div className={'paragraph_desc'}>
                    <h1 className={'title_description'} id="Clustering">Clustering</h1>
                    <p>
                        Clustering pipeline description to go in here

                </p>
                </div>
                <div className={'paragraph_desc'}>
                    <h1 className={'title_description'} id="Cell-type assignment">Cell-type assignment</h1>
                    <p>
                        For the cell type assignment we have currently used Azimuth from <a href={'https://satijalab.org/azimuth/'}>https://satijalab.org/azimuth/</a>. 
                        Azimuth is a web application that uses an annotated reference dataset to automate the processing, analysis, and interpretation of a new single-cell RNA-seq experiment.
                </p>
                </div>

            </div>

        )
    }
}

export default Descriptions
